Configure IDE with sbt server

Duhemm opened the pull request:
- When there's no sbt project, offer to create a new sbt project:
  - `build.sbt`
  - `project/build.properties`
  - `project/plugins.sbt`
- Try to connect to a running sbt server instance or start a new server.
  - If vscode cannot connect to sbt, ask the user to try again.
- Run `configureIDE` using `sbt/exec`
  - Report error when it fails, asking the user to try again
- Periodically check that sbt is still alive; start a new instance otherwise.

smarter commented on vscode-dotty/src/extension.ts:
```diff
@@ -19,7 +19,16 @@ let extensionContext: ExtensionContext
+const sbtCheckIntervalMs = 10 * 1000
+
+/** A command that we use to check that sbt is still alive. */
+export const nopCommand = "nop"
```
I'm not sure this is the way to do things:
- If the connection with sbt is closed, we don't need this since we can have a callback on connection.onClose
- If sbt blocks while running something, then we will never get a response since there's no timeout for a response
- We can't add a timeout for the response, because an sbt command can take an arbitrarily long time to finish (e.g. running tests) and sbt cannot execute more than one command at once.

I think a real solution will require changes to sbt, but I don't know what exactly.

Duhemm commented:
Remove `nop`. It's now relying on `connection.close` to reconnect.

Duhemm commented:
@smarter Everything seems to work on Windows now!

smarter commented on vscode-dotty/src/extension.ts:
```diff
@@ -43,47 +68,139 @@ export function activate(context: ExtensionContext) {
     }
+
+    configuredProject
+      .then(_ => connectToSbt(coursierPath))
```
Since starting sbt and running configureIDE might be very slow, I suggest we do not do it if the project has already been configured once (i.e., .dotty-ide-artifact, .dotty-ide.json are present). In the future, the Dotty Language Server will decide itself if it should query the build tool for configuration options.

Duhemm commented:
@smarter I removed some of the stuff that dealt with sbt server (but we're still using it). 
b01854bf9ef91f279d854befe90cfc871e88b71f is the commit to revert if/when we decide to revive that.

Duhemm commented on vscode-dotty/src/extension.ts:
```diff
@@ -43,47 +68,122 @@ export function activate(context: ExtensionContext) {
+  const failMessage = "`configureIDE` failed (see log for details)"
+
+  // `configureIDE` is a command, which means that upon failure, sbt won't tell us anything
+  // until sbt/sbt#4370 is fixed.
```
This has been fixed, and I think that the fix is in sbt 1.2.6

Duhemm commented:
> Why {} instead of void ?

Right, I changed it.

> If disableDottyIDEFile then the project is not configured, but we should not try to configure it either and just quit the extension, the logic to do this seems to have been lost.

Argh, thanks, I fixed it too.

smarter commented on vscode-dotty/src/extension.ts:
```diff
@@ -83,7 +82,7 @@ export function activate(context: ExtensionContext) {
-    if (isUnconfiguredProject()) {
+    if (!isConfiguredProject()) {
       configuredProject = vscode.window.showInformationMessage(
         "This looks like an unconfigured Scala project. Would you like to start the Dotty IDE?",
```
Looks like this message is still going to be shown if disableDottyIDEFile is present

Duhemm replied:
🤦‍♂️
